packages/component/src/providers/ActivityTree/ActivityTreeComposer.tsx

```diff
@@ -25,22 +25,28 @@ const ActivityTreeComposer = ({ children }: ActivityTreeComposerProps) => {
   const [rawActivities] = useActivities();
   const getActivitiesByKey = useGetActivitiesByKey();
   const getKeyByActivity = useGetKeyByActivity();
+  const activityKeys = useActivityKeys();
 
   const activities = useMemo<readonly WebChatActivity[]>(() => {
     const activities: WebChatActivity[] = [];
 
+    if (!activityKeys) {
```

Comment:
Shouldn't the call below `getKeyByActivity()` would return `undefined`?

Reply:
We do this to ensure we update the memo when activityKeys got updated. I should've added a comment on this.

Reply:
Oh wow, this is good. 👍🏻